pkg/cmd/pulumi/state_edit.go

```diff
@@ -248,12 +249,25 @@ func (cmd *stateEditCmd) validateAndPrintState(f *snapshotBuffer) (*deploy.Snaps
 }
 
 func openInEditor(filename string) error {
-	editor := os.Getenv("EDITOR")
-	if editor == "" {
+	editorEnv := os.Getenv("EDITOR")
+	if editorEnv == "" {
 		return fmt.Errorf("no EDITOR environment variable set")
 	}
+	return openInEditorInternal(editorEnv, filename)
+}
+
+func openInEditorInternal(editorEnv, filename string) error {
```

Comment:
```suggestion
func openInEditorInternal(editor, filename string) error {
```

Just call it editor, it's not important that it came from the envvar